internal/pkg/cli/pipeline_delete.go

```diff
@@ -88,21 +89,26 @@ func (o *deletePipelineOpts) Validate() error {
 		return errNoAppInWorkspace
 	}
 
-	if err := o.readPipelineManifest(); err != nil {
-		return err
+	if o.name != "" {
+		if _, err := o.codepipeline.GetPipeline(o.name); err != nil {
+			return err
+		}
 	}
-
 	return nil
 }
 
 // Ask prompts for fields that are required but not passed in.
 func (o *deletePipelineOpts) Ask() error {
+	if err := o.getNameAndSecret(); err != nil {
```

Comment:
It is pretty strange to parse and set these values in `Ask`, I'd have expected the parsing to happen in `Execute`.

Not in this PR! But I think it might be worthwhile moving all this logic into `Execute` instead 🤔 and having `Ask` here just call `PipelineSelector` for deployed pipelines. Do you agree?